src/calculateRank.js

```diff
@@ -1,105 +1,64 @@
-/**
- * Calculates the probability of x taking on x or a value less than x in a normal distribution
- * with mean and standard deviation.
- *
- * @see https://stackoverflow.com/a/5263759/10629172
- *
- * @param {string} mean The mean of the normal distribution.
- * @param {number} sigma The standard deviation of the normal distribution.
- * @param {number} to The value to calculate the probability for.
- * @returns {number} Probability.
- */
-const normalcdf = (mean, sigma, to) => {
-  var z = (to - mean) / Math.sqrt(2 * sigma * sigma);
-  var t = 1 / (1 + 0.3275911 * Math.abs(z));
-  var a1 = 0.254829592;
-  var a2 = -0.284496736;
-  var a3 = 1.421413741;
-  var a4 = -1.453152027;
-  var a5 = 1.061405429;
-  var erf =
-    1 - ((((a5 * t + a4) * t + a3) * t + a2) * t + a1) * t * Math.exp(-z * z);
-  var sign = 1;
-  if (z < 0) {
-    sign = -1;
-  }
-  return (1 / 2) * (1 + sign * erf);
-};
+function expsf(x, lambda = 1) {
+  return 2 ** (-lambda * x);
+}
 
 /**
  * Calculates the users rank.
  *
  * @param {object} params Parameters on which the user's rank depends.
- * @param {number} params.totalRepos Total number of repos.
- * @param {number} params.totalCommits Total number of commits.
- * @param {number} params.contributions The number of contributions.
- * @param {number} params.followers The number of followers.
+ * @param {boolean} params.all_commits Whether `include_all_commits` was used.
+ * @param {number} params.commits Number of commits.
  * @param {number} params.prs The number of pull requests.
  * @param {number} params.issues The number of issues.
- * @param {number} params.stargazers The number of stars.
+ * @param {number} params.repos Total number of repos.
+ * @param {number} params.stars The number of stars.
+ * @param {number} params.followers The number of followers.
  * @returns {{level: string, score: number}}} The users rank.
  */
-const calculateRank = ({
-  totalRepos,
-  totalCommits,
-  contributions,
-  followers,
+function calculateRank({
+  all_commits,
+  commits,
   prs,
   issues,
-  stargazers,
-}) => {
-  const COMMITS_OFFSET = 1.65;
-  const CONTRIBS_OFFSET = 1.65;
-  const ISSUES_OFFSET = 1;
-  const STARS_OFFSET = 0.75;
-  const PRS_OFFSET = 0.5;
-  const FOLLOWERS_OFFSET = 0.45;
-  const REPO_OFFSET = 1;
-
-  const ALL_OFFSETS =
-    CONTRIBS_OFFSET +
-    ISSUES_OFFSET +
-    STARS_OFFSET +
-    PRS_OFFSET +
-    FOLLOWERS_OFFSET +
-    REPO_OFFSET;
-
-  const RANK_S_VALUE = 1;
-  const RANK_DOUBLE_A_VALUE = 25;
-  const RANK_A2_VALUE = 45;
-  const RANK_A3_VALUE = 60;
-  const RANK_B_VALUE = 100;
+  repos, // unused
+  stars,
+  followers,
+}) {
+  const COMMITS_MEAN = all_commits ? 500 : 100, COMMITS_WEIGHT = 1;
+  const PRS_MEAN = 50, PRS_WEIGHT = 2;
+  const ISSUES_MEAN = 10, ISSUES_WEIGHT = 1;
```

Comment:
Issues now has the same weight as commits 🤔?

Reply:
Fixed ;)